jbmc/src/java_bytecode/java_types.h

```diff
@@ -242,6 +242,32 @@ inline bool can_cast_type<java_class_typet>(const typet &type)
   return can_cast_type<class_typet>(type);
 }
 
+class java_method_typet : public code_typet
+{
+public:
+  const std::vector<irept> &throws_exceptions() const
```

Comment:
I strongly dislike returning `vector<irept>` as caller knows literaly nothing about what those `irept`s might be, could be `symbol_typet` as in `bases()` on a class, or just strings as they actually are. To hide the implementation detail, I suggest changing to:

```C++
const std::vector<irep_idt> &throws_exceptions() const
{
  std:vector<irep_idt>exceptions;
  std::transform(
    find(ID_exceptions_thrown_list).get_sub().being(), 
    find(ID_exceptions_thrown_list).get_sub().end(), 
    irept::idt);
  return exceptions;
}
```

There may (must??) be a nicer way of doing this without the copy, but to me even this is preferable to an interface that returns essentially a typeless array. 

I'd also try and get rid of the setter, and instead take a `std::vector<irep_idt>` in the constructor so it is always in a valid state.

Reply:
OK I see the constructor part is not possible without some restructuring. Instead I suggest a `add_throws_exception` method that adds to the sub, since that is the only way anyone should be modifying it. (by giving the whole sub tree away, any caller could inadvertently remove all the exceptions).